### DISCUSSION TITLE
How to check if save file exists/has all needed variables?

### DISCUSSION THREAD

**Discussion by NikRetaNCAM** [HaxeFlixel/flixel#2801] [Q&A] [answered]
My game uses multiple save files (3),
```hx
override public function create(){
saves = [save1, save2, save3];
		save1.bind("save1");
		save2.bind("save2");
		save3.bind("save3");
}
```
They seem to work fine BUT in order for them to do that you need to manually reset the save
```hx
function resetSave()
	{
		saves[curSave].data.location = 'hub';
		saves[curSave].data.level = 0;

		saves[curSave].data.cardLv = 0;
		saves[curSave].data.money = 0;

		// MONEY 1, MONEY 2, MONEY 3, SECRET, TIME, DEATHS, RANK
		var lvlInfo:Map<String, Array<Dynamic>> = ["tut" => [false, false, false, false, null, null, null]];
		saves[curSave].data.lvlInfo = lvlInfo;

		saves[curSave].flush();
	}
```
otherwise the game will crash
What I want to do is to make this process automatic. So that when loading a fresh new save it will set itself up automatically. Is it possible to check the save for variables.. or.. smth?
Thx!

**Comment by Geokureli** [comment]
I just null check properties in the save data, if they are null the save must not exist

**Reply by NikRetaNCAM** [reply]
> I just null check properties in the save data, if they are null the save must not exist

so.. i shall check for every variable in the save and if it is null then the save shall reset itself, right?

**Reply by Geokureli** [reply, marked as answer]
no, i check if a specific var is null, and if it is, I know there is no save and set up the default save, or call resetSave in your case. for instance you can check for `saves[curSave].data.location == null` or something

**Reply by NikRetaNCAM** [reply]
> no, i check if a specific var is null, and if it is, I know there is no save and set up the default save, or call resetSave in your case. for instance you can check for `saves[curSave].data.location == null` or something

that helped. thanks!
